packages/lucia-auth/src/auth/index.ts

```diff
@@ -115,8 +115,14 @@ export class Auth<C extends Configuration = any> {
 		} else {
 			this.adapter = config.adapter(LuciaError);
 		}
-		this.generateUserId =
-			config.generateCustomUserId ?? (() => generateRandomString(15));
+		this.generateUserId = async () => {
+			let generator = config.generateCustomUserId ?? (() => generateRandomString(15));
+			let userId = await generator();
+			if (userId.includes(".")) {
+				throw new LuciaError("AUTH_INVALID_USER_ID");
```

Comment:
Hm, I don't think it's wise to use the same error message. Maybe use a different prefix (`AUTH_`)?

Reply:
Done, it's `GEN_INVALID_USER_ID` now.